MiguelAValdesG/SpringDataJPA/src/main/java/com/bootcamp/springdatajpa/service/impl/AddressServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+package com.bootcamp.springdatajpa.service.impl;
+
+import com.bootcamp.springdatajpa.dao.AddressDAO;
+import com.bootcamp.springdatajpa.dao.OwnerDAO;
+import com.bootcamp.springdatajpa.dto.AddressDTO;
+import com.bootcamp.springdatajpa.entity.Address;
+import com.bootcamp.springdatajpa.entity.Owner;
+import com.bootcamp.springdatajpa.exception.InvalidDataException;
+import com.bootcamp.springdatajpa.mapper.AddressMapper;
+import com.bootcamp.springdatajpa.service.AddressService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.bootcamp.springdatajpa.utils.Constants.ADDRESS_NOT_FOUND;
+import static com.bootcamp.springdatajpa.utils.Constants.ADDRESS_CANNOT_BE_DELETED;
+
+@Service
+public class AddressServiceImpl implements AddressService {
+
+  @Autowired
+  AddressDAO addressDAO;
+  @Autowired
+  OwnerDAO ownerDAO;
+
+  @Autowired
+  AddressMapper addressMapper;
+
+  @Override
+  public List<AddressDTO> getAllAddresses() {
+    return addressDAO.findAll().parallelStream()
+      .map(addressDAO -> addressMapper.addressEntityToDTO(addressDAO))
+      .collect(Collectors.toList());
+  }
+
+  @Override
+  public AddressDTO addAddress(AddressDTO addressDTO) {
+    return addressMapper.addressEntityToDTO(addressDAO.save(addressMapper.addressDTOToEntity(addressDTO)));
+  }
+
+  @Override
+  public AddressDTO updateAddress(Long id, AddressDTO addressDTO) {
+    return addressMapper.addressEntityToDTO(addressDAO.save(fillAddress(findAddress(id), addressDTO)));
+  }
+
+  @Override
+  public AddressDTO deleteAddress(Long id) {
+    Optional<Owner> ownerOptional = ownerDAO.findByIdAddress(id);
+    AddressDTO addressDTO;
+
+    if (!ownerOptional.isPresent()) {
+      Address address = findAddress(id);
+      addressDAO.delete(address);
+      addressDTO = addressMapper.addressEntityToDTO(address);
```

Comment:
I don't get it.... if you delete address in line 56... how is that you try to map this variable to a DTO in line 57 ???

HINT: CrudRepository.delete() is VOID

Reply:
Well, I did it like that because I wanted to respond to my client with the entity that he deleted. But you're rigth, this doesn't has sense.

FIXED!! Thank you!